examples/Web/embed_external_rois.html

```diff
@@ -0,0 +1,426 @@
+                };
+                if (viewport.loadedImg.tiles) {
+                    options['tiles'] = true;
+                }
+
+                viewport.viewportimg.roi_display(options);
+                viewport.viewportimg.get(0).setRoiZoom(viewport.viewportimg.get(0).getZoom());
+            }
+            else
+                console.log("Viewport already loaded");
+        };
+
+        var get_random_int = function (min_value, max_value) {
+            return Math.floor(Math.random() * (max_value - min_value)) + min_value;
+        }
+
+        var get_random_point = function (min_x, max_x, min_y, max_y) {
+            var px = get_random_int(min_x, max_x);
+            var py = get_random_int(min_y, max_y);
+
+            return {"x": px, "y": py};
+        }
+
+        var get_random_color = function () {
+            var colors = [
+                "red",
+                "green",
+                "blue",
+                "yellow",
+                "orange",
+                "olive",
+                "navy",
+                "pink"
+            ];
+
+            return colors[Math.floor(Math.random() * colors.length)];
+        };
+
+        var flush_colors = function () {
+            var vimg = viewport.viewportimg.get(0);
+            var ext_rois = vimg.get_external_roi_json();
+
+            for(var r=0; r<ext_rois.length; r++) {
+                var roi_id = ext_rois[r].id;
+                var shapes = ext_rois[r].shapes;
+                for (var s=0; s<shapes.length; s++) {
+                    var shape_id = shapes[s].id;
+                    var shape_color = shapes[s].fillColor;
+                    var new_color = get_random_color();
+                    while (new_color == shape_color) {
+                        new_color = get_random_color();
+                    }
+                    vimg.update_shape_config(roi_id, shape_id, new_color, undefined,
+                            undefined, new_color, undefined, false);
+                }
+            }
+
+            vimg.refresh_active_rois();
+        }
+
+        var get_img_properties = function () {
+            return {
+                "z_pos": viewport.getPos().z,
+                "t_pos": viewport.getPos().t,
+                "img_width": viewport.getSizes().width,
+                "img_height": viewport.getSizes().height
+            };
+        };
+
+        var add_random_shape = function (shape) {
+            if (!viewport.viewportimg.get(0).show_rois) {
+                load_viewport();
+            }
+
+            var roi_id;
+            if (shape.type == "Rectangle")
+                roi_id = "EXT_RECT_" + shapes_counter.rectangle;
+            else if (shape.type == "Ellipse")
+                roi_id = "EXT_ELL_" + shapes_counter.ellipse;
+            else if (shape.type == "Line")
+                roi_id = "EXT_LINE_" + shapes_counter.line;
+            viewport.viewportimg.get(0).push_shape(roi_id, 1, shape, true);
```

Comment:
At least in this example, all your shapes have the same ID, which breaks the clicking selection of shapes (single click selects all shapes, which then can't be unselected). Maybe just use a random number here, or something else just to get this working.